plugins/network-elements/nuage-vsp/src/com/cloud/network/manager/NuageVspManagerImpl.java

```diff
@@ -109,12 +163,49 @@
     @Inject
     VpcDao _vpcDao;
     @Inject
-    NuageVspDao nuageVspDao;
+    VpcManager _vpcManager;
+    @Inject
+    NuageVspDao _nuageVspDao;
+    @Inject
+    NuageVspSync _nuageVspSync;
+    @Inject
+    DataCenterDao _dataCenterDao;
+    @Inject
+    ConfigurationDao _configDao;
+    @Inject
+    NetworkModel _ntwkModel;
+    @Inject
+    AccountManager _accountMgr;
+    @Inject
+    IPAddressDao _ipAddressDao;
+    @Inject
+    FirewallRulesDao _firewallDao;
+    @Inject
+    VpcServiceMapDao _vpcSrvcDao;
+    @Inject
+    AgentManager _agentMgr;
+    @Inject
+    private DomainDao _domainDao;
     @Inject
-    NuageVspSync nuageVspSync;
+    NetworkOfferingDao _networkOfferingDao;
+    @Inject
+    NetworkOfferingServiceMapDao _networkOfferingServiceMapDao;
 
     private ScheduledExecutorService scheduler;
 
+    @Inject
+    MessageBus _messageBus;
+
+    static {
+        NUAGE_VSP_PROVIDERS = Sets.newHashSet(Network.Provider.NuageVsp);
```

Comment:
Use an ``ImmutableSet`` to ensure that the constant cannot be accidentally modified.